superfocus_app/do_alignment.py

```diff
@@ -78,22 +84,40 @@ def align_reads(query, output_dir, aligner, database, evalue, threads, fast_mode
     blast_mode = 'blastp' if amino_acid == '1' else 'blastx'
 
     if aligner == "diamond":
-        temp_folder = Path("{}/db/tmp/".format(WORK_DIRECTORY))
-
-        if not temp_folder.exists():
-            temp_folder.mkdir(parents=True, mode=511)
-
-        mode_diamond = "" if fast_mode == "1" else "--sensitive"
+        #mode_diamond = "" if fast_mode == "1" else "--sensitive"
         database_diamond = "{}/db/static/diamond/{}.db".format(WORK_DIRECTORY, database)
 
         # align
-        os.system("diamond {} -t {} -d {} -q {} -a {} -p {} -e {} {}".format(blast_mode, temp_folder, database_diamond,
-                                                                             query, output_name, threads, evalue,
-                                                                             mode_diamond))
+        # os.system("diamond {} -t {} -d {} -q {} -a {} -p {} -e {} {}".format(blast_mode, temp_folder, database_diamond,
+        #                                                                     query, output_name, threads, evalue,
+        #                                                                     mode_diamond))
+        diamond_blast = [
+            "diamond", blast_mode,
+            "-d", database_diamond,
+            "-q", query,
+            "-o", f"{output_name}.m8",
+            "-f", "6",
+            "-t", temp_folder,
+            "-p", threads,
+            "-e", evalue,
+        ]
+        if fast_mode != "1":
+            diamond_blast.append("--sensitive")
+        try:
+            retcode = subprocess.call(diamond_blast)
+            if retcode != 0:
+                print("Diamond blast was terminated by signal", retcode, file=sys.stderr)
+                sys.exit(retcode)
+        except OSError as e:
+            print("Diamond blast execution failed:", e, file=sys.stderr)
+            sys.exit()
+
+        # note that this is no longer a two step process
```

Comment:
please clean up commented lines. we can have the record from the older versions.